HunterPie/Features/Native/NativeIPCInitializer.cs

```diff
@@ -1,58 +1,22 @@
-﻿using HunterPie.Core.Address.Map;
-using HunterPie.Core.Game;
-using HunterPie.Core.Game.Rise;
-using HunterPie.Core.Logger;
+﻿using HunterPie.Core.Logger;
 using HunterPie.Core.Native.IPC;
-using HunterPie.Core.Native.IPC.Handlers.Internal.Initialiaze;
-using HunterPie.Domain.Interfaces;
-using System;
-using System.Linq;
 using System.Threading.Tasks;
 
 namespace HunterPie.Features.Native
 {
-    internal class NativeIPCInitializer : IContextInitializer
+    internal static class NativeIPCInitializer
     {
 
-        private static readonly string[] _addresses = new[]
+        public static async Task WaitForIPCInitialization()
         {
-            "FUN_CALCULATE_ENTITY_DAMAGE"
-        };
-
-        public void Initialize(Context context)
-        {
-            if (context is not MHRContext)
-                return;
-
-            IPCService.Initialize().ContinueWith(async (success) =>
+            var attempts = 0;
+            while (!(await IPCService.Initialize()))
             {
-                bool connected = await success;
-                
-                if (!connected)
-                {
-                    for (int i = 1; i <= 10; i++)
-                    {
-                        Log.Debug($"Retrying to connect: Attempt {i}...");
-
-                        connected = await IPCService.Initialize();
-
-                        if (connected)
-                            break;
-
-                        await Task.Delay(i * 100);
-                    }
-                }
-
-                UIntPtr[] addresses = GetAddressesToHook();
-
-                IPCInitializationMessageHandler.RequestIPCInitialization(addresses);
-            });
+                attempts++;
+                Log.Debug($"Retrying to connect: Attempt {attempts}...");
+                await Task.Delay(attempts * 100);
+            }
```

Comment:
The Native Interface isn't a required thing for HunterPie to work, therefore it doesn't need to wait for the IPC forever in an infinite loop. We can drop the connection initialization after 10 unsuccesful tries.